src/Data/Textile/Fabric.elm

```diff
@@ -83,34 +83,54 @@ fromString string =
             Err <| "Procédé de tissage/tricotage inconnu: " ++ string
 
 
-getMakingComplexity : MakingComplexity -> Fabric -> MakingComplexity
-getMakingComplexity productDefaultMakingComplexity fabric =
-    case fabric of
-        KnittingFullyFashioned ->
-            MakingComplexity.VeryLow
-
-        KnittingIntegral ->
-            MakingComplexity.NotApplicable
-
-        _ ->
-            productDefaultMakingComplexity
-
-
-getMakingWaste : Split -> Fabric -> Split
-getMakingWaste productDefaultWaste fabric =
-    case fabric of
-        KnittingFullyFashioned ->
-            Split.fromFloat 0.02
-                |> Result.toMaybe
-                |> Maybe.withDefault productDefaultWaste
-
-        KnittingIntegral ->
-            Split.fromFloat 0
-                |> Result.toMaybe
-                |> Maybe.withDefault productDefaultWaste
-
-        _ ->
-            productDefaultWaste
+getMakingComplexity : MakingComplexity -> Maybe MakingComplexity -> Maybe Fabric -> MakingComplexity
+getMakingComplexity defaultComplexity maybeCustomComplexity maybeFabric =
+    case ( maybeFabric, maybeCustomComplexity ) of
+        ( _, Just customComplexity ) ->
+            -- A custom complexity is provided: always takes priority
+            customComplexity
+
+        ( Just fabric, Nothing ) ->
+            -- Specific fabric process provided: retrieve associated complexity
+            case fabric of
+                KnittingFullyFashioned ->
+                    MakingComplexity.VeryLow
+
+                KnittingIntegral ->
+                    MakingComplexity.NotApplicable
+
+                _ ->
+                    defaultComplexity
+
+        ( Nothing, Nothing ) ->
+            -- No specific fabric process or complexity provided: fallback to defaults
+            defaultComplexity
```

Comment:
I know I'm nitpicking, but I can't help myself :sweat_smile: Feel free to ignore :nerd_face: 
```suggestion
        ( fabric, Nothing ) ->
            case fabric of
                -- Specific fabric process provided: retrieve associated complexity
                Just KnittingFullyFashioned ->
                    MakingComplexity.VeryLow

                Just KnittingIntegral ->
                    MakingComplexity.NotApplicable

                -- No specific fabric process or complexity provided: fallback to defaults
                _ ->
                    defaultComplexity

```